src/EditorFeatures/CSharpTest/NavigateTo/NavigateToTests.cs

```diff
@@ -1169,6 +1167,73 @@ public void ToError(__arglist)
     VerifyNavigateToResultItem(item, "ToError", "[|ToError|](__arglist)", PatternMatchKind.Exact, NavigateToItemKind.Method, Glyph.MethodPublic);
 });
         }
+
+        [Fact]
+        public async Task DoNotIncludeTrivialPartialContainer()
+        {
+            using var workspace = TestWorkspace.Create(@"
+<Workspace>
+    <Project Language=""C#"" CommonReferences=""true"">
+        <Document FilePath=""File1.cs"">
+            public partial class Outer
+            {
+                public void VisibleMethod() { }
+            }
+        </Document>
+        <Document FilePath=""File2.cs"">
+            public partial class Outer
+            {
+                public partial class Inner { }
+            }
+        </Document>
+    </Project>
+</Workspace>
+", composition: EditorTestCompositions.EditorFeatures);
+
+            _provider = new NavigateToItemProvider(workspace, AsynchronousOperationListenerProvider.NullListener);
+            _aggregator = new NavigateToTestAggregator(_provider);
+
+            VerifyNavigateToResultItems(
+                new()
+                {
+                    new NavigateToItem("Outer", NavigateToItemKind.Class, "csharp", null, null, s_emptyExactPatternMatch, null),
+                },
+                await _aggregator.GetItemsAsync("Outer"));
+        }
+
+        [Fact]
+        public async Task DoIncludeNonTrivialPartialContainer()
```

Comment:
❔ Can you describe the behavior of two-level (or more) nested types? Tests would be good for non-obvious cases.

Reply:
Effectively, a type is considered 'trivial' (and not included in the index) if it is partial, has members, and *all* those members are nested types.  This applies to a type regardless of if the type itself is an outer type or a nested type.

The general intuition here is that such types don't really arise in practice *except* to be a way to have nested types broken out from teh 'main' outer type.  And, as suh, we don't really consider those a valuable 'navigate to' target for a user.  in other words, given files:

Foo.cs
```cs
partial class Foo
{
     // main Foo bits
}
```

and Foo.Bar.cs:
```c#
partial class Foo
{
    class Bar
    {
        // Bar bits.
    }
}
```

The user doesn't think of the 'Foo' in Foo.Bar.cs as being significant (in terms of navigation).  It exists merely to ensure 'Bar' is properly contained.  So only navigable symbols in Foo.Bar.cs is just `class Bar` (and whatever is inside of that).